src/runtimes/node/detect_es_module.js

```diff
@@ -0,0 +1,22 @@
+const fs = require('fs')
+
+const { init, parse } = require('es-module-lexer')
```

Comment:
We already have a JavaScript parser ([`acorn`](https://www.npmjs.com/package/acorn)) as a dependency, which should be able to tell us whether a file uses ESM or not. If it is, I would try to use it before adding a new dependency.

Reply:
sure sure, I'll take a look 👍

Reply:
I am wondering about the performance difference there considering:
  - `acorn` is implemented in JavaScript while `es-module-lexer` is in WebAssembly
  - `acorn` does lexing+parsing while `es-module-lexer` does lexing only
  - `acorn` parses all possible JavaScript syntax while `es-module-lexer` only looks for imports/exports

The author of `es-module-lexer` actually [claims the following](https://github.com/guybedford/es-module-lexer#es-module-lexer):

> For an example of the performance, Angular 1 (720KiB) is fully parsed in 5ms, in comparison to the fastest JS parser, Acorn which takes over 100ms.